server/controllers/CommentController.js

```diff
@@ -0,0 +1,191 @@
+      where: { id, ownerId },
+      attributes: {
+        exclude: ['authorId'],
+      },
+      fields: updateFields,
+      returning: true,
+    })
+      .then(([updated, comment]) => {
+        if (!updated) {
+          return res.status(404)
+            .json({
+              errors: {
+                message: 'Comment not found, or you do '
+                  + 'not have permission to update this comment',
+              },
+            });
+        }
+        return res.status(200).json({ comment: comment[0] });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Get all the comments of an article
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @param {Object} next - a call to the next function
+   * @returns {Object} an array of comments
+   * on success and error message on error
+   */
+  static getComments(req, res, next) {
+    const { slug } = req.params;
+    Article.findOne({ where: { slug } })
+      .then((article) => {
+        errorHelper.throwError(article, 'Article not found', 404);
+        const articleId = article.id;
+        // query the database to retrieve comments alongside their reply count
+        return sequelize
+          .query(queryHelper.getComments(articleId), {
+            type: sequelize.QueryTypes.SELECT,
+          });
+      })
+      .then((rawComments) => {
+        const comments = commentCtrlHelper.parseComments(rawComments);
+        const commentCount = comments.length;
+        return res.status(200)
+          .json({ comments, commentCount });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Get all the replies to a comment
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @param {Object} next - a call to the next function
+   * @returns {Object} an array of comments on
+   * success and error message on error
+   */
+  static getReplies(req, res, next) {
+    const { slug, id: parentId } = req.params;
+    Article.findOne({ where: { slug } })
+      .then((article) => {
+        errorHelper.throwError(article, 'Article not found', 404);
+        const articleId = article.id;
+        // query the database to retrieve comments alongside their reply count
+        return sequelize
+          .query(queryHelper.getReplies(articleId, parentId), {
+            type: sequelize.QueryTypes.SELECT,
+          });
+      })
+      .then((rawComments) => {
+        const comments = commentCtrlHelper.parseComments(rawComments);
+        const commentCount = comments.length;
+        return res.status(200)
+          .json({ comments, commentCount });
+      })
+      .catch(next);
+  }
+
+  /**
+   * Delete a comment
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @param {Object} next - a call to the next function
+   * @returns {Object} status code 200 on success and error message on error
+   */
+  static deleteComment(req, res, next) {
```

Comment:
since you are not actually deleting the comments, don't you think `hideComment` will be more appropriate than `deleteComment` for the method name?